File: slicing-android/app/src/main/java/com/crescentflare/slicinggame/infrastructure/physics/PhysicsObject.kt
package com.crescentflare.slicinggame.infrastructure.physics

import android.graphics.PointF
import android.graphics.RectF
import com.crescentflare.slicinggame.infrastructure.geometry.Vector

/**
 * Physics: an object in the physics engine
 */
interface PhysicsObject {

    var x: Float
    var y: Float
    var lethal: Boolean
    var recursiveCheck: Int
    val collisionBounds: RectF
    val collisionRotation: Float
    val collisionPivot: PointF

    fun onCollision(hitObject: PhysicsObject?, normal: Vector, timeRemaining: Float, physics: Physics)

}

File: slicing-ios/SlicingGame/Infrastructure/Physics/PhysicsObject.swift
//
//  PhysicsObject.swift
//  Physics: an object in the physics engine
//

import UIKit

protocol PhysicsObject: class {

    var x: Float { get set }
    var y: Float { get set }
    var recursiveCheck: Int { get set }
    var collisionBounds: CGRect { get }
    var collisionRotation: Float { get }
    var collisionPivot: CGPoint { get }
    
    func didCollide(withObject: PhysicsObject?, normal: Vector, timeRemaining: TimeInterval, physics: Physics)

}
